cap/01-categorie.tex

```diff
@@ -1,9 +1,272 @@
 \chapter{Categorie, funtori, trasformazioni naturali}
+
+\section{Categorie}\label{categorie}
+
+Prima di arrivare alla definizione formale di categorie, può essere di aiuto avere in mente un esempio concreto. Prendiamo per esempio gli spazi vettoriali e le mappe lineari tra di loro. 
+Molti di voi sapranno che le matrici invertibili su $\ictR^n$ con il loro prodotto formano un gruppo. Più un generale, se tralasciamo la condizione di invertibilità, abbiamo un \emph{monoide}:
+
+\begin{definition}
+ Un \emph{monoide} consiste in un insieme $M$ con:
+ \begin{itemize}
+  \item Un elemento particolare $e\in M$, chiamato \emph{elemento neutro}, \emph{unità} o \emph{identità};
+  \item Un'operazione binaria $M\times M\to M$, chiamata \emph{prodotto} o \emph{composizione}, e indicata con $(m,n),\mapsto m\cdot n$;
+  che soddisfano le seguenti proprietà:
+  \item \emph{Unitalità}: per ogni $m\in M$, i prodotti $m \cdot e$ ed $e\cdot m$ sono uguali a $m$;
+  \item \emph{Associatività}: per ogni $m,n,p\in M$, i prodotti $(m\cdot n)\cdot p$ ed $m\cdot (n\cdot p)$ sono uguali. 
+ \end{itemize}
+\end{definition}
+
+Se ora consideriamo gli spazi vettoriali $\ictR^n$ per diversi $n$, non tutte le mappe lineari si possono comporre. In particolare, una mappa $f:\ictR^n\to \ictR^m$ si può comporre con una mappa $g:\ictR^p\to\ictR^q$ solo se $m=p$. In altre parole, $g$ si può comporre con $f$ solo se il codominio di $f$ è uguale al dominio di $g$.
```

Comment:
La notazione per le funzioni/mappe/morfismi non dovrebbe usare `\colon`, come in `f \colon A \to B`? Questo è un problema di stile generale, lo noto qui solo perché è il primo punto dove mi ci è saltato l'occhio.

Reply:
Corretto.